src/pages/scheduler/components/ShareTimetableButton.tsx

```diff
@@ -0,0 +1,185 @@
+import { CopyIcon } from '@chakra-ui/icons';
+import { Heading, HStack, VStack, Text, WrapItem, Wrap } from '@chakra-ui/layout';
+import {
+  Alert,
+  AlertIcon,
+  Box,
+  Button,
+  Flex,
+  Icon,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+  useMediaQuery,
+  useToast,
+} from '@chakra-ui/react';
+import { HiLink } from 'react-icons/hi';
+import { IoShareOutline } from 'react-icons/io5';
+import { useParams } from 'react-router';
+import {
+  EmailIcon,
+  EmailShareButton,
+  FacebookIcon,
+  FacebookShareButton,
+  TelegramIcon,
+  TelegramShareButton,
+  TwitterIcon,
+  TwitterShareButton,
+  WhatsappIcon,
+  WhatsappShareButton,
+} from 'react-share';
+
+import { SavedCourse, useSavedCourses } from 'lib/hooks/useSavedCourses';
+import { getReadableTerm } from 'lib/utils';
+
+const InformationText = (props: { isSmallScreen: boolean; term: string }) => {
+  return (
+    <Alert status="info" borderRadius="10px">
+      {props.isSmallScreen ? <AlertIcon /> : null}
+      We've generated a link that you can share to allow people to view, compare, and import the courses and sections
+      you currently have selected for the {getReadableTerm(props.term)} term.
+    </Alert>
+  );
+};
+
+const ShareCourseCard = (props: { course: SavedCourse }) => {
```

Comment:
The way we structure props in custom components is usually like such
```suggestion
type Props = {
   course: SavedCourse;
}
const ShareCourseCard = ({ course }: Props) => {
```
this is called object de-structuring, which is generally nicer because we don't have to call `props.` everytime, and can reference the variables directly